explainer.md

```diff
@@ -121,6 +121,106 @@ The currently proposed API differs in a number of aspects:
   recent 'sync' meeting.
 
 ## Examples
+
+The new APIs, in their most basic form:
+```js
+const example = `<b onclick="alert(1)">hello world</b>`;
+const element = document.createElement("div");
+
+// Modify element:
+element.setHTML(example);  // <div><b>hello world</b></div>
+element.setHTMLUnsafe(example);  // <div><b onclick="alert(1)">hello world</b></div>
+
+// Return a Document instance:
+Document.parseHTML(example);  // <html><head></head><body><b>hello world</b></body></html>
+Document.parseHTMLUnsafe(example);  // <html><head></head><body><b onclick="alert(1)">hello world</b></body></html>
+```
+
+Parsing observes its contexts:
+```js
+const example_tr = `<tr><td>A table row.</td></tr>`;
+const table = document.createElement("table");
+
+element.setHTML(example_tr);  // <div>A table row.</div>
+table.setHTML(example_tr);  // <table><tbody><tr><td>A table row.</td></tr></tbody></table>
+Document.parseHTML(example_tr);  // <html><body>A table row.</body></html>
+                                 // (Left out <head> for clarity.)
+```
+All of these would have had identical results if the "unsafe" variants had
+been used.
+
+Parsing follows HTML parsing rules, unlike `innerHTML`, where it depends on the
+document type:
+```js
+const element_xml = new DOMParser().parseFromString("<html><body><div/></body></html>", "application/xhtml+xml").getElementsByTagName("div")[0];
+const example_not_xml = "<bLoCkQuOtE>bla";
+
+element_xml.getRootNode().contentType;  // 'application/xhtml+xml'
+element_xml.innerHTML = example_not_xml;  // Throws.
+element_xml.setHTML(example_not_xml);  // <div><blockquote>bla</blockquote></div>
+                                       // Note case and closing elements.
+element.setHTML(example_not_xml);  // Same as above.
+```
+
+The "safe" methods remove all script-y content defined by the platform and
+let the rest pass:
+```js
+element.setHTML(`<a href=about:blank onclick=alert(1) onload=alert(2) id=myid class=something><script>alert(3);</script>`);
+// <div><a href="about:blank" id="myid" class="something"></a></div>
+```
+
+The operation of the built-in sanitizer can be configured to suit your
+applications' needs. Both "safe" and "unsafe" versions can take a configuration.
+(Please note that naming and structure here is rather preliminary,
+but we expect these capabilities to be in the final standard.)
+
+The "safe" version will ignore configuration items that break its security
+guarantees:
+```js
+const an_unsafe_config = { 'allowElements': [ { name: 'script' } ] };
+element.setHTML("<script>", { sanitizer: an_unsafe_config });  // <div></div>
+element.setHTMLUnsafe("<script>", { sanitizer: an_unsafe_config });  // You now have a script. Congrats.
+```
+
+For elements, the HTML namespace is default. For attributes, the empty namespace.
```

Comment:
null namespace. Empty string is normalized to null. Also applies below.

Reply:
Done.